### DISCUSSION TITLE
Custom parameter info

### DISCUSSION THREAD

**Discussion by noman-khan1224** [eclipse-langium/langium#1504] [Q&A]
The current node description is shown as:(property) Reference<AstNode>.$nodeDescription?: AstNodeDescription | undefined
I want to replace this with a custom description if it is available, and have this custom description inserted into the AstNode.

![Screenshot (5)](https://github.com/eclipse-langium/langium/assets/161299971/d13ac17c-5207-4949-a777-315aa00fe829)

Request you to help with an approach for same?

**Comment by msujew** [comment]
Hey @noman-khan1224,

Just so I understand you correctly: You want to enrich the existing `$nodeDescription` property on the `Reference` interface with additional properties?

You would do that by overriding the [`AstNodeDescriptionProvider#createDescription`](https://github.com/eclipse-langium/langium/blob/75aa73e001848a5aaeb92e1648e1f07d6e7064f6/packages/langium/src/workspace/ast-descriptions.ts#L23-L37) method and returning a new interface that extends the existing `AstNodeDescription`. Afterwards, once the reference is linked, the contained `$nodeDescription` will be of the type that you returned from your `createDescription` override. In order to use that type in TypeScript, you will need to upcast:

```ts
const nodeDescription = ref.$nodeDescription as CustomNodeDescription | undefined;
```

**Reply by noman-khan1224** [reply]
The FieldMember displays the type of the entity. However, if a description is available for an entity in the AstNode, I want to display the custom description instead of the FieldMember.

![image](https://github.com/eclipse-langium/langium/assets/161299971/4af4e38c-acc0-4ac4-89d3-12fa632b1888)

Request you to help with an approach for same?

**Reply by cdietrich** [reply]
see [`DefaultCompletionProvider.createReferenceCompletionItem`](https://github.com/eclipse-langium/langium/blob/6971c83afb924336d70b3e3092440478c73ac92e/packages/langium/src/lsp/completion/completion-provider.ts#L460-L467) it calls `detail: nodeDescription.type,`
so you may want to customize there
